codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -728,11 +729,45 @@ picToObj' pic =
             return $ (unsafeCoerce arr :: JSVal)
     setProps xs obj = liftIO $ void $ mapM (\(s, v) -> setProp s v obj) xs
 
+-- describePicture 
+describePicture :: Picture -> String
+describePicture (Rectangle _ w h) = printf "rectangle { width = %4f , height = %4f }" w h
+describePicture (SolidPolygon _ pts) = printf "solidPolygon " ++ intercalate "," [printf "(%4f, %4f)" x y | (x,y) <- pts]
```

Comment:
It would be great to stick to a consistent format, by adding braces and a name for the parameter.  Something like this:

    describePicture (SolidPolygon _ pts) = printf "solidPolygon { points = %s }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts])

The same comment applies to other cases where you do something similar.